### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by vektor8891**
Changelist:
- #22 vector slicing
- #23 add unit test to PLHashMap functions
- #31 extend join-with to numbers
- #33 calculate product of number list
- also reverted changes from commit 3bae54a

### CODE REVIEW

File: src/data/hashMap/PLHashMap.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export class PLHashMap<Item extends PLBase> implements PLBase, Index<PLString, I
   }
 
   public toJS(): Map<string, any> {
-    return new Map(Array.from(this._value.entries()).map(([k, v]) => [k, v.toJS()]))
+    return this._value
```
Comment (maxinteger) [inline src/data/hashMap/PLHashMap.ts:32]: why did you change the `toJS` implementation?

File: src/data/hashMap/PLHashMap.spec.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ describe('stdlib/data/PLHashMap', () => {
     it('should return with the JS representation', () => {
       const js = plHashMap(plString('a'), plNumber(1), plString('b'), plNumber(2)).toJS()
       expect(Array.from(js.keys())).toEqual(['a', 'b'])
-      expect(Array.from(js.values())).toEqual([1, 2])
+      expect(Array.from(js.values())).toEqual([plNumber(1), plNumber(2)])
```
Comment (maxinteger) [inline src/data/hashMap/PLHashMap.spec.ts:23]: `toJS` must applied recursively to the items of the hashMap
if it is return with plNumber then it is wrong

**Review by maxinteger — CHANGES_REQUESTED** [review]

File: src/data/hashMap/PLHashMap.ts (hunk 1 above)
Comment (vektor8891) [reply src/data/hashMap/PLHashMap.ts]: @maxinteger It wasn't intentional. I reverted the changes.

File: src/data/hashMap/PLHashMap.spec.ts (hunk 2 above)
Comment (vektor8891) [reply src/data/hashMap/PLHashMap.spec.ts]: @maxinteger It wasn't intentional. I reverted the changes.

**Review by maxinteger — CHANGES_REQUESTED** [review]
Something is not good here.
The code loos OK, but on of the check return with error and if I run locally the tests one of them fails:
```
stdlib/data/PLHashMap › toJS › should return with the JS representation
```

**Comment by vektor8891** [comment]
@maxinteger It is fixed now.

**Comment by maxinteger** [comment]
Sorry @vektor8891, my bad.
I forget to upload some changes to the repository.
Could you please update your branch one more time?

**Comment by vektor8891** [comment]
@maxinteger No worries, here it is. Unfortunately, the tests failed but I think it wasn't my code. Also, package-lock.json is also added now. Is it ok?

**Comment by vektor8891** [comment]
@maxinteger Can you pls have a look? After running `npm update pocket-lisp` all tests pass.

**Comment by maxinteger** [comment]
@vektor8891 Tests are still failing on the CI server. But looks good locally.
I guess CI might cached an old version of Pocket lisp and that cause the issue.

Please open `.travis.yml` and remove this line:
```
cache: npm
```
than commit and push the changes

**Comment by vektor8891** [comment]
@maxinteger I've removed the cache but it still doesn't seem to work.

File: src/data/vector/vectorFn.ts [hunk 3]
```diff
@@ -27,9 +27,9 @@ export const join: (list: PLVector<PLString>) => PLString = (list) => {
   return list.reduce(plString(''), add)
 }
 
-export const joinWith: (separator: PLString, list: PLVector<PLString>) => PLString = (separator, list) => {
+export const joinWith: (separator: PLString, list: PLVector<PLString|PLNumber>) => PLString = (separator, list) => {
```
Comment (maxinteger) [inline src/data/vector/vectorFn.ts:30]: this is wrong, we should not allow implicit type conversation in any PL functions

here if the user pass number it will be implicitly converted into string

**Review by maxinteger — CHANGES_REQUESTED** [review]

File: src/data/vector/vectorFn.ts (hunk 3 above)
Comment (vektor8891) [reply src/data/vector/vectorFn.ts]: @maxinteger I've created a separate function for numbers. Can you please take a look?

**Comment by vektor8891** [comment]
@maxinteger Is there anything I can do for this PR? If not, can you pls merge?
